src/agentscope/utils/common.py

```diff
@@ -26,6 +26,9 @@ def timer(seconds: Optional[Union[int, float]] = None) -> Generator:
     Note:
         This function only works in Unix,
         since `signal.setitimer` is only available in Unix.
+
+        This function may cause error when not used in main thread,
```

Comment:
Please add `TODO` here.
And can we print some warning when using timer with agentscope_studio?